distil/primitives/fuzzy_join.py

```diff
@@ -0,0 +1,545 @@
+            ) from error
+
+        accuracy = self.hyperparams["accuracy"]
+        if accuracy <= 0.0 or accuracy > 1.0:
+            raise exceptions.InvalidArgumentValueError(
+                "accuracy of " + str(accuracy) + " is out of range"
+            )
+
+        left_col = self.hyperparams["left_col"]
+        right_col = self.hyperparams["right_col"]
+
+        # perform join based on semantic type
+        join_type = self._get_join_semantic_type(
+            left, left_resource_id, left_col, right, right_resource_id, right_col
+        )
+        joined: pd.Dataframe = None
+        if join_type in self._STRING_JOIN_TYPES:
+            joined = self._join_string_col(
+                left_df, left_col, right_df, right_col, accuracy
+            )
+        elif join_type in self._NUMERIC_JOIN_TYPES:
+            joined = self._join_numeric_col(
+                left_df, left_col, right_df, right_col, accuracy
+            )
+        elif join_type in self._VECTOR_JOIN_TYPES:
+            joined = self._join_vector_col(
+                left_df, left_col, right_df, right_col, accuracy
+            )
+        elif join_type in self._DATETIME_JOIN_TYPES:
+            joined = self._join_datetime_col(
+                left_df, left_col, right_df, right_col, accuracy
+            )
+        else:
+            raise exceptions.InvalidArgumentValueError(
+                "join not surpported on type " + str(join_type)
+            )
+
+        # create a new dataset to hold the joined data
+        resource_map = {}
+        float_vector_columns = {}
+        for resource_id, resource in left.items():  # type: ignore
+            if resource_id == left_resource_id:
+                for column in joined.columns:
+                    # need to avoid bug in container.Dataset, it doesn't like vector columns
+                    if type(joined[column][0]) == np.ndarray:
+                        float_vector_columns[column] = joined[column]
+                        joined[column] = np.NAN
+                resource_map[resource_id] = joined
+            else:
+                resource_map[resource_id] = resource
+        result_dataset = container.Dataset(resource_map, generate_metadata=True)
+        for key in float_vector_columns.keys():
+            df = result_dataset["0"]
+            df[key] = float_vector_columns[key]
+            float_vec_loc = df.columns.get_loc(key)
+            float_vec_col_indices = df.metadata.list_columns_with_semantic_types(
+                ("https://metadata.datadrivendiscovery.org/types/FloatVector",)
+            )
+            if float_vec_loc not in float_vec_col_indices:
+                df.metadata = df.metadata.add_semantic_type(
+                    (metadata_base.ALL_ELEMENTS, float_vec_loc),
+                    "https://metadata.datadrivendiscovery.org/types/FloatVector",
+                )
+
+        return base.CallResult(result_dataset)
+
+    def multi_produce(
```

Comment:
this can  be removed - `multi_produce` has been deprecated

Reply:
Sounds like we still need it, so we'll have to figure out what the suggested alternative since its been deprecated.